src/crush/CrushCompiler.cc

```diff
@@ -314,10 +314,12 @@ int CrushCompiler::decompile(ostream &out)
 
   out << "\n# devices\n";
   for (int i=0; i<crush.get_max_devices(); i++) {
-    out << "device " << i << " ";
-    print_item_name(out, i, crush);
-    print_item_class(out, i, crush);
-    out << "\n";
+    const char *name = crush.get_item_name(i);
```

Comment:
This prints device name. But how it skips device id ..

Reply:
```get_item_name()``` returns 0 if ```i``` is not found in the crushmap. So the following line only prints the devices that are actually in the crushmap.